src/poliastro/twobody/propagation.py

```diff
@@ -188,6 +195,868 @@ def _kepler(k, r0, v0, tof, *, numiter):
+    if ecc >= 1.0:
+        raise ValueError("Mikkola only works with elliptical orbits.")
+    M0 = nu_to_M(nu, ecc)
+    a = p / (1 - ecc ** 2)
+    n = np.sqrt(k / a ** 3)
+    M = M0 + n * tof
+
+    if ecc < 1.0:
+        # Solving elliptical case
+
+        # Equation (9a)
+        alpha = (1 - ecc) / (4 * ecc + 1 / 2)
+        beta = M / 2 / (4 * ecc + 1 / 2)
+
+        # Equation (9b)
+        if beta >= 0:
+            z = (beta + np.sqrt(beta ** 2 + alpha ** 3)) ** (1 / 3)
+        else:
+            z = (beta - np.sqrt(beta ** 2 + alpha ** 3)) ** (1 / 3)
+
+        # Equation (9c)
+        s = z - alpha / z
+
+        # Apply correction term
+        ds = -0.078 * s ** 5 / (1 + ecc)
+        s = s + ds
+
+        # Equation (8)
+        E = M + ecc * (3 * s - 4 * s ** 3)
+        nu = E_to_nu(E, ecc)
+
+        return coe2rv(k, p, ecc, inc, raan, argp, nu)
+
+
+def markley(k, r, v, tofs, rtol=None):
+    """ Kepler Equation solver based on a fifth-order refinement of the solution
+    of a cubic equation.
+
+    Parameters
+    ----------
+    k : ~astropy.units.Quantity
+        Standard gravitational parameter of the attractor.
+    r : ~astropy.units.Quantity
+        Position vector.
+    v : ~astropy.units.Quantity
+        Velocity vector.
+    tofs : ~astropy.units.Quantity
+        Array of times to propagate.
+    rtol: float
+        This method does not require of tolerance since it is non iterative.
+
+    Returns
+    -------
+    rr : ~astropy.units.Quantity
+        Propagated position vectors.
+    vv : ~astropy.units.Quantity
+        Propagated velocity vectors.
+
+    Note
+    ----
+    This method was originally presented by Markley in his paper *Kepler Equation Solver*
+    with DOI: https://doi.org/10.1007/BF00691917
+    """
+
+    k = k.to(u.m ** 3 / u.s ** 2).value
+    r0 = r.to(u.m).value
+    v0 = v.to(u.m / u.s).value
+    tofs = tofs.to(u.s).value
+
+    results = [_markley(k, r0, v0, tof) for tof in tofs]
+    return (
+        [result[0] for result in results] * u.m,
+        [result[1] for result in results] * u.m / u.s,
+    )
+
+
+def _markley(k, r0, v0, tof):
+    """ Solves the kepler problem by a non iterative method. Relative error is
+    around 1e-18, only limited by machine double-precission errors.
+
+    Parameters
+    ----------
+    k : float
+        Standar Gravitational parameter
+    r0 : array
+        Initial position vector wrt attractor center.
+    v0 : array
+        Initial velocity vector.
+    tof : float
+        Time of flight.
+
+    Returns
+    -------
+    rf: array
+        Final position vector
+    vf: array
+        Final velocity vector
+
+    Note
+    ----
+    The following algorithm was taken from http://dx.doi.org/10.1007/BF00691917.
+    """
+
+    # Solve first for eccentricity and mean anomaly
+    p, ecc, inc, raan, argp, nu = rv2coe(k, r0, v0)
+    if ecc >= 1.0:
+        raise ValueError("Markley only works with elliptical orbits.")
```

Comment:
In this case it seems true that this propagator only claims to work for eccentric orbits

Reply:
However,

> but the eccentricity can be arbitrarily close to this limit

So perhaps we should check `if ecc > 1.0` instead?